TEQ-48 Move celerybeat PID file to /var/run

vkurup opened the pull request:
closes #27

vkurup committed:
Move celerybeat PID file to /var/run

jbradberry reviewed (approved):
:ice_cream:

dsummersl commented on tasks/celery.yml:
```diff
@@ -7,7 +7,7 @@
     - { name: "default", command: "worker", flags: "{{ celery_worker_extra_args|default('--loglevel=INFO') }}" }
-    - { name: "beat", command: "beat", flags: "--schedule={{ root_dir }}/celerybeat-schedule.db --pidfile={{ root_dir }}/celerybeat.pid --loglevel=INFO" }
+    - { name: "beat", command: "beat", flags: "--schedule={{ root_dir }}/celerybeat-schedule.db --pidfile=/var/run/celerybeat.pid --loglevel=INFO" }
 
```
The service doesn't start with these changes. Celery wants  to be run by the project owner, not root, and /var/run/* is owned by root.

Bopping around the internet it looks like the standard way to do this is to create a directory owned by the service in /var/run, owned by the service's user, and to put the pid in that directory. But since /var/run is a temporary filesystem the directory must be created on startup. My suggestion would be:

1. Create a /etc/tmpfile.d/celery.conf rule to create /var/run/celery, owned by the project's user at boot time (and rerun systemd-tmpfile --create when it is installed).
2. Change the line above to put the pidfile into /var/run/celery/celerybeat.pid

I've made these changes on my local tequila-django version of this branch, and that works across reboots. If that sounds like a reasonable change I'll push them up to this branch.

dsummersl replied:
secret gist: 
https://gist.github.com/dsummersl/e4ecf35250272e13b00c3475bc8c1d30

dsummersl committed:
Put celery PIDs in /var/run/celery (TEQ-48)

PID ownership is transfered from root to project_owner.

This change will use systemd-tmpfiles to ensure that /var/run/celery is
created whenever the server starts.

vkurup reviewed (commented):
Github won't let me 'approve' my own PR ;)

This looks good to me. :+1: Thanks @dsummersl

dsummersl committed:
Adds explanatory docs to systemd-tmpfiles settings for celery (TEQ-27)

dsummersl committed:
Fixes typo.

dsummersl commented:
Neil has verified this on the craftcarejoy project.
